http/jsonapi/runtime/parameters.go

```diff
@@ -94,18 +100,27 @@ func ScanParameters(w http.ResponseWriter, r *http.Request, parameters ...*ScanP
 			panic(fmt.Errorf("impossible scanning location: %d", param.Location))
 		}
 
-		n, _ := fmt.Sscan(scanData, param.Data) // nolint: gosec
+		n, _ := Scan(scanData, param.Data)
 		// only report on non empty data, govalidator will handle the other cases
 		if n != 1 && scanData != "" {
-			WriteError(w, http.StatusBadRequest, &Error{
-				Title:  fmt.Sprintf("invalid value for %s", param.Name),
-				Detail: fmt.Sprintf("invalid value, expected %s got: %q", reflect.ValueOf(param.Data).Type(), scanData),
-				Source: &map[string]interface{}{
-					"parameter": param.Name,
-				},
-			})
+			WriteError(w, http.StatusBadRequest, param.BuildInvalidValueError(reflect.ValueOf(param.Data).Type(), scanData))
 			return false
 		}
 	}
 	return true
 }
+
+// Scan works like fmt.Sscan except for strings, they are directly assigned
+func Scan(str string, data interface{}) (int, error) {
+	// Don't handle plain strings with sscan but directly assign the value
+	t := reflect.TypeOf(data)
+	if t.Kind() == reflect.Ptr && t.Elem().Kind() == reflect.String {
+		strPtr, ok := data.(*string)
+		if ok {
```

Comment:
The result of both `if`s cannot be different, right? So you could have only the inner `if` to reduce complexity.